include/IECore/Version.h

```diff
@@ -0,0 +1,97 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2020, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+//     * Neither the name of Image Engine Design nor the names of any
+//       other contributors to this software may be used to endorse or
+//       promote products derived from this software without specific prior
+//       written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef IECORE_VERSION_H
+#define IECORE_VERSION_H
+
+#define CORTEX_MILESTONE_VERSION IE_CORE_MILESTONEVERSION
```

Comment:
This might be a bit confusing, so I'll try to explain

- I preferred `CORTEX_` for the #defines as those will get used in external repos and I think its our long long-term preference for naming.
- I avoided the `!VERSION!` style that Gaffer is using so other header/source files within Cortex can make use of Version.h as well.
- I do `sed` out the `IE_CORE_` bits when installing/deploying Version.h so for external repos those are baked down to ints

Let me know if that's too confusing or acceptable...

Reply:
> I preferred CORTEX_* for the #defines as those will get used in external repos and I think its our _long-term preference for naming_.

I'm not sure about this bit, although it may even be me who suggested it. `IE_CORE` is the "namespace" for macros in the same way that `namespace IECore` is the namespace for everything else. Do you anticipate changing the C++ namespaces too? I think it may be more pragmatic to stick with the original namespaces. There's precedent for this in OpenEXR, where the project name is facility-agnostic, but the internal namespaces reflect the original heritage.

That said, I'm sure there's an argument that this one set of macros should be `CORTEX_`, because it refers to the version of the whole package, not just the one library.

Reply:
I went to look at the OpenEXR example to see what they've done in this respect. Their external facing VERSION macros all use the newer project name (eg `OPENEXR_VERSION_STRING`, `OPENEXR_VERSION_MAJOR`, etc). I'm still tempted to follow that pattern, but I don't feel strongly enough to debate further if you'd prefer I switch back to `IE_CORE_MAJOR_VERSION`. Just let me know.

Reply:
I don't feel strongly either - let's stick with `CORTEX_` then.